Corona-Warn-App/src/androidTest/java/de/rki/coronawarnapp/datadonation/analytics/modules/exposurewindows/AnalyticsExposureWindowsDatabaseTest.kt

```diff
@@ -0,0 +1,76 @@
+package de.rki.coronawarnapp.datadonation.analytics.modules.exposurewindows
+
+import androidx.room.Room
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import io.kotest.matchers.shouldBe
+import kotlinx.coroutines.runBlocking
+import org.junit.After
+import org.junit.Test
+import org.junit.runner.RunWith
+import testhelpers.BaseTestInstrumentation
+
+@RunWith(AndroidJUnit4::class)
+class AnalyticsExposureWindowsDatabaseTest : BaseTestInstrumentation() {
+
+    private val database: AnalyticsExposureWindowDatabase =
+        Room.inMemoryDatabaseBuilder(
+            ApplicationProvider.getApplicationContext(),
+            AnalyticsExposureWindowDatabase::class.java
+        )
+            .build()
+
+    private val dao = database.analyticsExposureWindowDao()
+
+    @After
+    fun teardown() {
+        database.clearAllTables()
+    }
+
+    @Test
+    fun testMoveToReportedAndRollback() = runBlocking {
+        //insert new
+        val exposureWindowEntity = AnalyticsExposureWindowEntity("hash", 1, 1, 1, 1, 1.0, 1)
+        val scanInstance = AnalyticsScanInstanceEntity(null, "hash", 1, 1, 1)
+        val wrapper = AnalyticsExposureWindowEntityWrapper(exposureWindowEntity, listOf(scanInstance))
+        dao.insert(listOf(wrapper))
+        val allNew = dao.getAllNew()
+        allNew.size shouldBe 1
+        allNew[0].exposureWindowEntity.sha256Hash shouldBe "hash"
+
+        // move to reported
+        dao.moveToReported(listOf(wrapper), 999999)
+        dao.getAllNew() shouldBe listOf()
+        val reported = dao.getReported("hash")
+        assert(reported != null)
+        reported?.sha256Hash shouldBe "hash"
```

Comment:
Subjective opinion, please don't use `?.` for values under test, just let it fail with `!!.`, while `assert` also ensures this, it doesn't add much to the readability of the test.